parser/types/field_type.go

```diff
@@ -77,6 +77,16 @@ func (ft *FieldType) IsDecimalValid() bool {
 	return true
 }
 
+// IsVarLengthType Determine whether the column type is a variable-length type
+func (ft *FieldType) IsVarLengthType() bool {
+	if ft.tp == mysql.TypeVarchar || ft.tp == mysql.TypeVarString ||
+		ft.tp == mysql.TypeJSON ||
+		ft.tp == mysql.TypeBlob || ft.tp == mysql.TypeTinyBlob || ft.tp == mysql.TypeMediumBlob || ft.tp == mysql.TypeLongBlob {
+		return true
+	}
+	return false
```

Comment:
What about numeric or decimal type?
What if the type is varchar(1)?

Reply:
Even varchar(1) is var length, but the max width is just 1. Should we consider it too?

Reply:
Hmm，I don't know whether it is good to performance even the varchar or char max length is 1. Let's just ignore them for now.